### ISSUE TITLE
Ability to see compound logs for test item

### ISSUE THREAD

**Issue by pskvortsov** [reportportal/reportportal#226]
Now test item contains only logs from current test, but this doesn't help when I need to look for methods that were run before test method.
For example, I want to see compound logs for each item, that will contain logs from BeforeClass, BeforeMethod and Test methods.

**Comment by DzmitryHumianiuk** [comment]
@pskvortsov i really love this idea.

just a question - which option is preferable?
*Option 1:* load and show logs of @Before* right above the logs of test item, following with specific line separator. The same for @After*

*Option 2:* show `Load` button right before test item logs, and show them only after click on this button?


CC @Iren-Melnyk

**Comment by dagansandler** [comment]
@DzmitryHumianiuk If I understand correctly, the difference between the 2 options you suggested is merely the existence of the "Load" button controlling collapse/expansion of the before/after logs. If that's the case I would personally prefer option 2, so I can control the scope of logs I'm looking at.

One more thing - I think that it may be needed in some cases to also see logs from @Test methods that were executed right before my failing test. Tests should always be independent, but alas, sometimes they  accidentally affect each other's flow/state.

**Comment by DzmitryHumianiuk** [comment]
@dagansandler good clarification. Thank you.

the only problem we see, that testing engines do not provide information, which before- or -after method belongs to current @test.

the only way we can make it, when we use start and finish time, to identify the sequence of execution.
But multi-threading ruins this approach.

**Comment by dagansandler** [comment]
Hmm, yes, that's a good point. I'm not sure how such feature would/should behave in parallel test executions.

An idea that pops to my mind as a possible alternative (but is probably out of scope for this feature request) is some sort of timeline view for the launch, where I can also see which tests/configuration methods ran in parallel to mine,

**Comment by DzmitryHumianiuk** [comment]
@dagansandler some sort of this?

![image](https://user-images.githubusercontent.com/11332788/43724717-597fbffe-99a3-11e8-8498-a9dc67662ba7.png)

**Comment by dagansandler** [comment]
Sort of, yeah!
I had more of a git branches type of timeline in my head, but this visualization might have richer information.
Is this from a feature already in the works?

**Comment by DzmitryHumianiuk** [comment]
not yet. but we grooming backlog to adjust priorities

**Comment by avyas** [comment]
We are using allure2 along side reportportal and timeline info in allure do present this info.

Would love to see some thing like this in reportportal as well.

**Comment by DzmitryHumianiuk** [comment]
@avyas we will add it as a part of #259
